camel/agents/role_assignment.py

```diff
@@ -0,0 +1,125 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+import re
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+from tenacity import retry, stop_after_attempt, wait_exponential
+
+from camel.agents import ChatAgent
+from camel.messages import BaseMessage
+from camel.prompts import TextPrompt
+from camel.typing import ModelType, RoleType
+
+
+class RoleAssignmentAgent(ChatAgent):
+    r"""
+    An agent that generates role names based on the task prompt.
+    Attributes:
+        role_assignment_prompt (TextPrompt): A prompt for the agent to generate
+        role names.
+    args:
+        model (ModelType): The tupe of model to use for the agent.
+            (default: :obj: 'ModelType.GPT_3_5_TURBO')
+        model_config (Any): The configuration for the model.
+            (default: :obj: 'None')
+    """
+
+    def __init__(
+        self,
+        model: ModelType = ModelType.GPT_3_5_TURBO,
+        model_config: Optional[Any] = None,
+    ) -> None:
+        self.role_assignment_prompt = TextPrompt(
+            'Given this task, "{task}", generate two role names, ' +
+            'one for the AI user and one for the AI assistant.')
+
+        system_message = BaseMessage(
+            role_name="Role Assigner",
+            role_type=RoleType.ASSISTANT,
+            meta_dict=None,
+            content="You assign roles based on tasks.",
+        )
+        super().__init__(system_message, model, model_config)
+
+    @retry(wait=wait_exponential(min=5, max=60), stop=stop_after_attempt(5))
+    def run_role_with_description(
+        self,
+        num_roles: Optional[int] = 2,
+        task_prompt: Union[str, TextPrompt] = "",
+    ) -> Tuple[List[str], Dict[str, str], bool, Dict[str, Any]]:
+        r""" "
+        Generate role names based on the input task prompt.
+
+        Args:
+            num_roles (int): The number of roles to generate.
+                (default: :obj:`2`)
+            task_prompt (Union[str, TextPrompt]): The prompt
+                for the task based on which the roles are to be generated.
```

Comment:
I guess the `task_prompt` is required. Should we put it as a positional argument?